plugins/processors/gpuattributes/internal/awsneuron_metric_modifier.go

```diff
@@ -0,0 +1,262 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package internal
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent/internal/containerinsightscommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+	"strings"
+)
+
+const (
+	aggregatedMetricSuffix = "_total"
+	ErrorType              = "error_type"
+	StatusType             = "status_type"
+	EventType              = "event_type"
+	logTypeSuffix          = "AWSNeuron"
+	MemoryLocation         = "memory_location"
+
+	Core                            = "Core"
+	Device                          = "Device"
+	Percentile                      = "percentile"
+	PodName                         = "PodName"
+	Count                           = "Count"
+	Bytes                           = "Bytes"
+	Seconds                         = "Seconds"
+	Percent                         = "Percent"
+	NeuronCoreAttributeKey          = "NeuronCore"
+	duplicateNeuronCoreAttributeKey = "neuroncore"
+	NeuronDeviceAttributeKey        = "NeuronDevice"
+	RuntimeTag                      = "runtime_tag"
+)
+
+type AwsNeuronMetricModifier struct {
+	logger *zap.Logger
+}
+
+type MetricModifications struct {
+	DuplicationTypes         []string
+	AttributeKeysToBeRemoved []string
```

Comment:
The `AttributeKeysToBeRemoved` part seems pretty fragile - what if Neuron Monitor starts returning new attributes that mess up the grouping again?  Would it be better to have an *allowlist* of all the attributes we know we want, so we can guarantee grouping, rather than a blocklist of extra attributes we want to remove?

Reply:
I see, I'll write it that way